packages/cli/src/oclif/commands/convert.js

```diff
@@ -5,72 +5,37 @@ const { callAPI } = require('../../utils/api');
-              `Integration ${appId} @ ${version} is already a CLI integration and can't be converted. Instead, pick a version that was created using the Visual Builder.`
-            );
-          }
-          this.stopSpinner();
-
-          return convertApp(
-            appInfo,
-            versionInfo.definition_override,
-            tempAppDir
+      // has info about the app, such as title
+      // has a CLI version of the actual app implementation
+      this.throwForInvalidVersion(version);
+      this.startSpinner('Downloading integration from Zapier');
+      try {
+        const [appInfo, versionInfo] = await Promise.all([
+          callAPI(`/apps/${appId}`, undefined, true),
+          callAPI(`/apps/${appId}/versions/${version}`, undefined, true),
+        ]);
+
+        if (!versionInfo.definition_override) {
+          this.error(
+            `Integration ${appId} @ ${version} is already a CLI integration and can't be converted. Instead, pick a version that was created using the Visual Builder.`
           );
-        } catch (e) {
-          if (e.status === 404) {
-            this.error(
-              `Visual Builder integration ${appId} @ ${version} not found. If you want to convert a Legacy Web Builder app, don't pass a \`--version\` option. Otherwise, double check the integration id and version.`
-            );
-          }
-          this.error(e);
         }
-      } else {
-        // has info about the app, such as title
-        const legacyDumpUrl = `${BASE_ENDPOINT}/api/developer/v1/apps/${appId}/dump`;
-        // has a CLI version of the actual app implementation
-        const cliDumpUrl = `${BASE_ENDPOINT}/api/developer/v1/apps/${appId}/cli-dump`;
-
-        this.startSpinner('Downloading integration from Zapier');
-
-        try {
-          const [legacyApp, appDefinition] = await Promise.all([
-            // these have weird call signatures because we're not calling the platform api
-            callAPI(null, { url: legacyDumpUrl }, true),
-            callAPI(null, { url: cliDumpUrl }, true),
-          ]);
-          // The JSON dump of the app doesn't have app ID, let's add it here
-          legacyApp.general.app_id = appId;
+        this.stopSpinner();
 
-          this.stopSpinner();
-
-          return convertApp(legacyApp, appDefinition, tempAppDir);
-        } catch (e) {
-          if (e.status === 404) {
-            this.error(
-              `Legacy Web Builder app ${appId} not found. If you want to convert a Visual Builder integration, make sure to pass a \`--version\` option.`
-            );
-          }
-          this.error(e);
+        return convertApp(appInfo, versionInfo.definition_override, tempAppDir);
+      } catch (e) {
+        if (e.status === 404) {
+          this.error(
+            `Visual Builder integration ${appId} @ ${version} not found. Double check the integration id and version.`
+          );
         }
+        this.error(e.json.errors[0]);
```

Comment:
this was the change I actually came in here to do, but figured I'd clean up while I was here anyway.

Reply:
Did you make this change because some developer was seeing an obscure error? If so, can you provide a before/after?